src/libopensc/card-openpgp.c

```diff
@@ -2451,23 +2630,34 @@ pgp_parse_and_set_pubkey_output(sc_card_t *card, u8* data, size_t data_len,
 
 		if (tag == 0x0081) {
 			/* set the output data */
-			if (key_info->rsa.modulus) {
-				memcpy(key_info->rsa.modulus, part, len);
+			if (key_info->u.rsa.modulus) {
+				memcpy(key_info->u.rsa.modulus, part, len);
```

Comment:
The memory allocated might be too small for the modulus generated with the current key.
I am aware that it is an "inherited" issue.

Reply:
see above